server/src/main/java/org/opensearch/search/pipeline/ProcessorExecutionDetail.java

```diff
@@ -0,0 +1,221 @@
+    public static final String PROCESSOR_EXECUTION_DETAILS_KEY = "processorExecutionDetails";
+
+    /**
+     * Constructor for ProcessorExecutionDetail
+     */
+    public ProcessorExecutionDetail(String processorName, long durationMillis, Object inputData, Object outputData) {
+        this.processorName = processorName;
+        this.durationMillis = durationMillis;
+        this.inputData = inputData;
+        this.outputData = outputData;
+    }
+
+    public ProcessorExecutionDetail(String processorName) {
+        this(processorName, 0, null, null);
+
+    }
+
+    public ProcessorExecutionDetail(StreamInput in) throws IOException {
+        this.processorName = in.readString();
+        this.durationMillis = in.readLong();
+        this.inputData = in.readGenericValue();
+        this.outputData = in.readGenericValue();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(processorName);
+        out.writeLong(durationMillis);
+        out.writeGenericValue(inputData);
+        out.writeGenericValue(outputData);
+    }
+
+    public String getProcessorName() {
+        return processorName;
+    }
+
+    public long getDurationMillis() {
+        return durationMillis;
+    }
+
+    public Object getInputData() {
+        return inputData;
+    }
+
+    public Object getOutputData() {
+        return outputData;
+    }
+
+    /**
+     * Adds or updates the input data for this processor execution detail.
+     *
+     * @param inputData the new input data
+     */
+    public void addInput(Object inputData) {
+        this.inputData = inputData;
+    }
+
+    /**
+     * Adds or updates the output data for this processor execution detail.
+     *
+     * @param outputData the new output data
+     */
+    public void addOutput(Object outputData) {
+        this.outputData = outputData;
+    }
+
+    /**
+     * Adds or updates the duration of the processor execution.
+     *
+     * @param durationMillis the new duration in milliseconds
+     */
+    public void addTook(long durationMillis) {
+        this.durationMillis = durationMillis;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(PROCESSOR_NAME_FIELD.getPreferredName(), processorName);
+        builder.field(DURATION_MILLIS_FIELD.getPreferredName(), durationMillis);
+        addFieldToXContent(builder, INPUT_DATA_FIELD.getPreferredName(), inputData, params);
+        addFieldToXContent(builder, OUTPUT_DATA_FIELD.getPreferredName(), outputData, params);
+        builder.endObject();
+        return builder;
+    }
+
+    private void addFieldToXContent(XContentBuilder builder, String fieldName, Object data, Params params) throws IOException {
+        if (data == null) {
+            builder.nullField(fieldName);
+            return;
+        }
+
+        if (data instanceof List) {
```

Comment:
We could also have a case of `data instanceof Map`. Moreover, did we try all the [core processors](https://opensearch.org/docs/latest/search-plugins/search-pipelines/search-processors/) to come up with such cases?